AppCenterData/AppCenterData/MSData.m

```diff
@@ -607,6 +573,73 @@ - (void)upsertFromCosmosDbWithPartition:(NSString *)partition
+  [self performCosmosDbOperationWithPartition:partition
+                                   documentId:nil
+                                   httpMethod:kMSHttpMethodGet
+                                     document:nil
+                            additionalHeaders:additionalHeaders
+                            additionalUrlPath:nil
+                            completionHandler:^(NSData *_Nullable data, NSHTTPURLResponse *_Nullable response,
+                                                NSError *_Nullable cosmosDbError) {
+                              // If not OK.
+                              if (response.statusCode != MSHTTPCodesNo200OK) {
+                                MSDataError *actualDataError = [MSCosmosDb cosmosDbErrorWithResponse:response
+                                                                                     underlyingError:cosmosDbError];
+                                MSLogError([MSData logTag],
+                                           @"Unable to list documents for partition %@: %@. Status code %ld when expecting %ld.", partition,
+                                           [actualDataError localizedDescription], (long)response.statusCode, (long)MSHTTPCodesNo200OK);
+                                MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:actualDataError
+                                                                                                    partition:partition
+                                                                                                 documentType:documentType];
+                                completionHandler(documents);
+                                return;
+                              }
+
+                              // Deserialize the list payload and try to get the array of documents.
+                              NSError *deserializeError;
+                              id jsonPayload = [NSJSONSerialization JSONObjectWithData:(NSData *)data options:0 error:&deserializeError];
+                              MSDataError *deserializeDataError = nil;
+                              if (!deserializeError && ![MSDocumentUtils isReferenceDictionaryWithKey:jsonPayload
+                                                                                                  key:kMSDocumentsKey
+                                                                                              keyType:[NSArray class]]) {
+                                deserializeDataError = [[MSDataError alloc] initWithErrorCode:MSACDataErrorJSONSerializationFailed
+                                                                                   innerError:deserializeError
+                                                                                      message:@"Can't deserialize documents"];
+                              }
+                              if (deserializeDataError) {
```

Comment:
Move the check inside the previous _if_. `deserializeDataError` can only be assigned inside the previous _if_.